File: 백준/Silver/11047. 동전 0/동전 0.py
# for문으로 돌면서 했더니 시간초과 발생

import sys
input = sys.stdin.readline

N, K = map(int, input().split())

coins = [int(input()) for _ in range(N)]
coins.reverse()

total = 0
count = 0


for coin in coins:
    if K < coin:
        continue

    else:
        count += K // coin
        K %= coin
    
print(count)

File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require("fs");
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
// const input = fs.readFileSync("./예제.txt").toString().split("\n");
let [N, K] = input[0].split(" ").map(Number);

const coins = [];
let count = 0;

for (let i = 1; i < N + 1; i++) {
  coins.push(parseInt(input[i]));
}

while (coins.length > 0) {
  const coin = coins.pop();

  if (K < coin) continue;

  count += Math.floor(K / coin);
  K %= coin;
}

console.log(count);
